flink-staging/flink-gelly/src/main/java/org/apache/flink/graph/example/AdamicAdarSimilarityMeasure.java

```diff
@@ -0,0 +1,243 @@
+
+		//vertices are given weights in graph
+		graph = graph.joinWithVertices(degrees,new AssignWeightToVertices());
+
+		//neighbors are computed for all the vertices
+		DataSet<Tuple2<Long, Tuple2<Double, HashSet<Tuple2<Long, Double>>>>> computedNeighbors =
+				graph.reduceOnNeighbors(new GatherNeighbors(), EdgeDirection.ALL);
+
+		//graph is updated with the new vertex values
+		Graph<Long, Tuple2<Double, HashSet<Tuple2<Long, Double>>>, Double> graphWithVertexValues =
+				graph.joinWithVertices(computedNeighbors, new UpdateGraphVertices());
+
+		//edges are given Adamic Adar coefficient as value
+		DataSet<Edge<Long, Double>> edgesWithAdamicValues = graphWithVertexValues.getTriplets().map(new ComputeAdamic());
+
+		// emit result
+		if (fileOutput) {
+			edgesWithAdamicValues.writeAsCsv(outputPath, "\n", ",");
+
+			// since file sinks are lazy, we trigger the execution explicitly
+			env.execute("Executing Adamic Adar Similarity Measure");
+		} else {
+			edgesWithAdamicValues.print();
+		}
+	}
+
+	@Override
+	public String getDescription() {
+		return "Adamic Adar Similarity Measure";
+	}
+
+	/*
+	 * Each vertex has a Tuple2 as value. The first field of the Tuple is the weight of the vertex
+	 * The second field is the HashSet containing of all the neighbors and their weights
+	 */
+	private static final class GatherNeighbors implements ReduceNeighborsFunction<Tuple2<Double,
+																HashSet<Tuple2<Long, Double>>>> {
+		@Override
+		public Tuple2<Double, HashSet<Tuple2<Long, Double>>>
+		reduceNeighbors(Tuple2<Double, HashSet<Tuple2<Long, Double>>> first,
+						Tuple2<Double, HashSet<Tuple2<Long, Double>>> second) {
+			first.f1.addAll(second.f1);
+			return first;
+		}
+	}
+
+	//mapfunction to initialise Vertex values from edge dataset
+	private static class mapVertices implements MapFunction<Long, Tuple2<Double, HashSet<Tuple2<Long, Double>>>> {
+		@Override
+		public Tuple2<Double, HashSet<Tuple2<Long, Double>>> map(Long value) {
+			HashSet<Tuple2<Long, Double>> neighbors = new HashSet<Tuple2<Long, Double>>();
+			neighbors.add(new Tuple2<Long, Double>(value, 0.0));
+			Double val = Double.MIN_VALUE;
+			return new Tuple2<Double, HashSet<Tuple2<Long, Double>>>(val, neighbors);
+		}
+	}
+
+	//mapfunction to assign weight for each vertex as 1/log(kn) where kn is the degree of the vertex
+	private static class AssignWeightToVertices implements
+			MapFunction<Tuple2<Tuple2<Double, HashSet<Tuple2<Long, Double>>>, Long>,
+					Tuple2<Double, HashSet<Tuple2<Long, Double>>>> {
+		@Override
+		public Tuple2<Double, HashSet<Tuple2<Long, Double>>>
+		map(Tuple2<Tuple2<Double, HashSet<Tuple2<Long, Double>>>, Long> value) throws Exception {
+
+			value.f0.f0 = 1.0/Math.log(value.f1);
+
+			for(Tuple2<Long, Double> t : value.f0.f1) {
+				t = new Tuple2<Long, Double> (t.f0, value.f0.f0);
+				value.f0.f1.clear();
+				value.f0.f1.add(t);
+			}
+			return value.f0;
+		}
+	}
+
+	//mapfunction to update values of vertices in the Graph
```

Comment:
Let's be consistent and out them all with /** ...*/ when they're in front of classes